code/scripting/api/objs/message.cpp

```diff
@@ -32,6 +32,25 @@ ADE_VIRTVAR(Name, l_Persona, "string", "The name of the persona", "string", "The
 	return ade_set_args(L, "s", Personas[idx].name);
 }
 
+ADE_VIRTVAR(Index, l_Persona, nullptr, nullptr, "number", "The index of the persona")
+{
+	int idx = -1;
+
+	if (!ade_get_args(L, "o", l_Persona.Get(&idx)))
+		return ade_set_args(L, "i", -1);
+
+	if (Personas.empty())
+		return ade_set_args(L, "i", -1);
+
+	if (!SCP_vector_inbounds(Personas, idx))
+		return ade_set_args(L, "i", -1);
+
+	if (ADE_SETTING_VAR)
+		LuaError(L, "Setting index is not supported");
+
+	return ade_set_args(L, "i", idx);
```

Comment:
Ah one last thing, the `mn.Personas` indexer uses a 1-based index so the `l_Persona` Index virtvar should also return a 1-based index. 

Similarly, the Index virtvars of `l_Submodel` and `l_Rank`, along with the `getShipClassIndex` function of `l_Shipclass` all return the Lua 1-based index. I'll plan to merge too once this one is updated to the 1-based index since we've given others enough time to look at it. :)